packages/contracts/src/SubdomainController.sol

```diff
@@ -2,45 +2,34 @@
 pragma solidity ^0.8.17;
 
 import {INameWrapper} from "@ens-contracts/wrapper/INameWrapper.sol";
+import {ENSRegistry} from "@ens-contracts/registry/ENSRegistry.sol";
 import {Resolver} from "@ens-contracts/resolvers/Resolver.sol";
 import {BytesUtils} from "@ens-contracts/utils/BytesUtils.sol";
 
 import {ENSHelper} from "../script/ENSHelper.sol";
-import {OffchainRegister} from "./interfaces/OffchainResolver.sol";
+import {
+    OffchainRegister,
+    RegisterRequest
+} from "./interfaces/OffchainResolver.sol";
 
 contract SubdomainController is OffchainRegister, ENSHelper {
 
     using BytesUtils for bytes;
 
     uint256 public price;
-    uint256 public commitTime;
     INameWrapper nameWrapper;
 
-    constructor(
-        address _nameWrapperAddress,
-        uint256 _price,
-        uint256 _commitTime
-    ) {
-        commitTime = _commitTime;
+    constructor(address _nameWrapperAddress, uint256 _price) {
         price = _price;
         nameWrapper = INameWrapper(_nameWrapperAddress);
     }
 
-    function register(
-        bytes calldata name,
-        address owner,
-        uint256 duration,
-        bytes32, /* secret */
-        address resolver,
-        bytes[] calldata data,
-        bool, /* reverseRecord */
-        uint16 fuses,
-        bytes memory /* extraData */
-    )
+    function register(RegisterRequest calldata request)
```

Comment:
Não seria bom emitir um evento de registro aqui dentro? A não ser que isso seja feito no setSubnodeRecord do wrapper.

Reply:
yep, events are emitted by the ENS Registry. this contract serves as a facade for our interface.